lib/mongodb.js

```diff
@@ -180,6 +184,8 @@ function MongoDB(settings, dataSource) {
   }
 
   this.dataSource = dataSource;
+  connectorSettings = this.settings;
```

Comment:
This isn't going to work. There can be multiple connector instances used by a single application (a single Node.js process).

Please find a different way how to access connector settings from functions that are not bound to the prototype.